### PR TITLE
Themes: Preview: Add "Try & Customize" button for modified sites

### PR DESCRIPTION
**Pull request by sirbrillig**
When #6349 was merged, it (intentionally) removed the "Try & Customize" button from both the Theme Sheet and the Theme Preview. Partially this was because if a site will be Headstarted (will have its content replaced by placeholder content) on theme switch, any changes made in the Customizer would actually prevent Headstart from running.

This PR restores the "Try & Customize" button as a *secondary* option in the preview, but only if the site would *not* be Headstarted. That is, sites with unmodified Headstart content will *not* see the button at all (although the option is still available from the ... menu in the showcase).

screen shot 2016-07-20 at 4 16 30 pm

### Testing

- Visit the theme showcase, use the ... menu to preview a theme. Nothing should be different.
- Visit the ThemeSheet and click the "open live demo" link.
- If your site has been recently created and has no modified content, you should NOT see the "Try & Customize" button
- If your site has modified content, you should see two buttons, including "Try & Customize"
- Try the above for the active theme, non-active theme, single-site, multi-site, and logged-out to be sure everything works as expected. 
- The only views that should appear different are the single-site and multi-site view for a non-active theme.

### CODE REVIEW

**Comment by folletto** [comment]
I wouldn't add the button on the Theme Sheet main page bar. I would add it inside the Preview, which is opened from both the list and the sheet. There's just not enough space on smaller screens and mobile to make that work.

**Comment by ockham** [comment]
> That is, sites with unmodified Headstart content will not see the button at all (although the option is still available from the ... menu in the showcase).

Do we want to hide it from there too? Part the idea of #6349 was to bundle display criteria for individual options in their `hideForTheme` and `hideForSite` attrs, so we could rather easily invoke the selector there.

**Comment by sirbrillig** [comment]
> I wouldn't add the button on the Theme Sheet main page bar. I would add it inside the Preview, which is opened from both the list and the sheet. 

Ok, I misunderstood; that's easier (halfway there: see screenshot above). However, that makes me wonder how useful this will be. My interpretation of the discussion that led to this button being restored was that users were frustrated that they couldn't customize before picking, which means that they weren't able to find the existing "Try & Customize" button in the theme showcase. I think that the theme preview is equally hidden. OTOH, maybe that's an issue for the theme preview button.

**Comment by folletto** [comment]
> However, that makes me wonder how useful this will be. My interpretation of the discussion that led to this button being restored was that users were frustrated that they couldn't customize before picking, which means that they weren't able to find the existing "Try & Customize" button in the theme showcase. I think that the theme preview is equally hidden. OTOH, maybe that's an issue for the theme preview button.

Agreed on all the instances, I just don't think the above is the right solution. Also we clarified the wording (to avoid erroneous activations) as well as the button is currently not present in the preview at all (it's just in the ••• menu).

So incrementally, adding both "Try" and "Activate" buttons in the live demo preview is a good step forward, even if I agree it might not be definitive. :)

**Comment by sirbrillig** [comment]
@ockham thanks! I didn't know about `hideForSite`. Perfect.

**Comment by sirbrillig** [comment]
Rebased to fix conflicts with #6844 and to squash changes.

**Comment by sirbrillig** [comment]
Added testing instructions.

**Comment by folletto** [comment]
Tested, design wise 👍 

I just noticed that if I open it from the theme sheets it shows both buttons (✓) but from the theme list it just shows Try & Customize. Is there a reason for that, or is it a bug?

**Comment by sirbrillig** [comment]
> but from the theme list it just shows Try & Customize. Is there a reason for that, or is it a bug?

For reference, here's the steps to get that button:

1. Visit the theme showcase (`/design`)
2. Click the ... and choose "Live Preview"

screen shot 2016-07-21 at 10 33 22 am


@folletto That's the behavior in `master` also, so it is unchanged. If it is not intentional, that might be outside the scope of this PR to fix.

**Comment by folletto** [comment]
> That's the behavior in master also, so it is unchanged. If it is not intentional, that might be outside the scope of this PR to fix (but should be made into an Issue if so!).

It's intentional in the sense that we haven't changed yet. :)

I thought the scope of this PR was to show the two buttons _everywhere_ the live demo preview shows up. Otherwise we'll need another PR to standardize the change that is introduced here, and make it consistent everywhere, yes.

But, if that's not the scope, then... what's the scope of this PR? Re-reading the original post:

> This PR restores the "Try & Customize" button as a secondary option in the preview, but only if the site would not be Headstarted. That is, sites with unmodified Headstart content will not see the button at all (although the option is still available from the ... menu in the showcase).

If that's the reason... I'm not sure. I think the two options should be shown every time, Headstart or not, since that gives the user the option to either start by customizing (thus skipping Headstart) or just previewing and activating (thus initiating Headstart where possibile).

In short, the change I think is to show everywhere, all the times, both "Try" and "Pick/Activate". 

If not... then I'm surely missing the rationale. I'm sorry for the poor clarity here.

**Comment by sirbrillig** [comment]
After some discussion in Slack, it seems I was confused about the idea behind this change. 😁 

The intent was *not* to only show "Try & Customize" when the site would not be Headstarted, but to show it all the time. And not only that, but also to add an "Activate" (or "Pick") button as the primary button when the primary button would otherwise be "Try & Customize" so that the UI is consistent. 

I'm going to make a new PR for this change because it's fairly different from this one and maybe we might want `isSiteContentUnmodified` later.

**Comment by sirbrillig** [comment]
New PR is #6952
